packages/edit-site/src/components/add-new-template/new-template.js

```diff
@@ -269,12 +270,13 @@ function useMissingTemplates(
 ) {
 	const existingTemplates = useExistingTemplates();
 	const defaultTemplateTypes = useDefaultTemplateTypes();
+	const extraTemplateTypes = useExtraTemplateTypes();
```

Comment:
Here we would need more checks. At least check if template slugs exist, do not allow override with core template slugs(`date, category`, etc..). We would still need to check if it plays well with the dynamic created templates(ex `custom-taxonomies`, etc..).

Reply:
Do you mean we should not allow overriding any template under `DEFAULT_TEMPLATE_SLUGS` with `extraTemplateTypes`? Something like this:
```js
const missingExtraTemplateTypes = ( extraTemplateTypes || [] ).filter(
	( template ) =>
		! DEFAULT_TEMPLATE_SLUGS.includes( template.slug ) &&
		! existingTemplateSlugs.includes( template.slug )
);
```

Or maybe better to check the extra template slugs don't override any other template previously added and do this at the end of the function? Like:
```js
const missingTemplateSlugs = ( missingTemplates || [] ).map(
	( { slug } ) => slug
);
const missingExtraTemplateTypes = ( extraTemplateTypes || [] ).filter(
	( template ) => ! missingTemplateSlugs.includes( template.slug )
);
return [ ...missingTemplates, ...missingExtraTemplateTypes ];
```

Reply:
Hey @ntsekouras I've pushed some changes to avoid extra templates overriding any of the other templates, let me know what you think 🙏